gateway-messages/src/mgs_to_sp.rs

```diff
@@ -322,14 +325,57 @@ pub enum MonorailComponentAction {
 pub enum UnlockChallenge {
     /// Unlock given an [UnlockResponse::Trivial] with the same timestamp
     Trivial { timestamp: u64 },
+
+    /// Hash and sign the given data with a key that the SP trusts
+    ///
+    /// Trusted keys are hardcoded into SP firmware.
+    EcdsaSha2Nistp256(EcdsaSha2Nistp256Challenge),
+}
+
+#[derive(
+    Copy,
+    Clone,
+    Serialize,
+    SerializedSize,
+    Deserialize,
+    PartialEq,
+    Eq,
+    Debug,
+    zerocopy::AsBytes,
+)]
+#[repr(C)]
+pub struct EcdsaSha2Nistp256Challenge {
+    /// Hardware ID, e.g. serial name
+    pub hw_id: [u8; 32],
+    /// Software version ID
+    pub sw_id: [u8; 4],
+    /// Time (according to the SP's internal clock)
+    pub time: [u8; 8],
+    /// Additional nonce chosen by the SP
+    pub nonce: [u8; 32],
 }
 
 /// Response to an [`UnlockChallenge`]
 #[derive(
     Copy, Clone, Serialize, SerializedSize, Deserialize, PartialEq, Eq, Debug,
 )]
 pub enum UnlockResponse {
+    /// Unlocks [UnlockChallenge::Trivial]
     Trivial { timestamp: u64 },
+
+    /// Here's your signature!
+    EcdsaSha2Nistp256 {
+        /// SEC1 encoding of the corresponding public key
+        #[serde(with = "BigArray")]
+        key: [u8; 65],
```

Comment:
It would be nice to see a symbolic constant, but a quick dive into the elliptic_curve::sec1 crate didn't reveal an answer to me.